tests/migrations/test_commands.py

```diff
@@ -2397,15 +2397,27 @@ def test_makemigrations_check(self):
         makemigrations --check should exit with a non-zero status when
         there are changes to an app requiring migrations.
         """
-        with self.temporary_migration_module() as tmpdir:
-            with self.assertRaises(SystemExit):
-                call_command("makemigrations", "--check", "migrations", verbosity=0)
-            self.assertFalse(os.path.exists(tmpdir))
+        out = io.StringIO()
+        with (
+            self.temporary_migration_module() as tmpdir,
+            self.assertRaises(SystemExit) as cm,
+        ):
+            call_command(
+                "makemigrations",
+                "--check",
+                "migrations",
+                stdout=out,
+            )
+        self.assertEqual(cm.exception.code, 1)
+        self.assertIn("Migrations for 'migrations':", out.getvalue())
+        self.assertFalse(os.path.exists(tmpdir))
 
+        out = io.StringIO()
```

Comment:
While I acknowledge that the current `main` have both checks in the same unit test, I would very much like us to use this opportunity to split the tests in two:
 * test_makemigrations_check_with_changes
 * test_makemigrations_check_no_changes

What do you think @adamchainz?

Reply:
Agree, I like to split unit tests.